v2/internal/frontend/desktop/windows/single_instance.go

```diff
@@ -0,0 +1,160 @@
+//go:build windows
+
+package windows
+
+import (
+	"encoding/json"
+	"github.com/wailsapp/wails/v2/internal/frontend/desktop/windows/winc/w32"
+	"github.com/wailsapp/wails/v2/pkg/options"
+	"golang.org/x/sys/windows"
+	"os"
+	"syscall"
+	"unicode/utf16"
+	"unsafe"
+)
+
+type COPYDATASTRUCT struct {
+	dwData uintptr
+	cbData uint32
+	lpData uintptr
+}
+
+// WMCOPYDATA_SINGLE_INSTANCE_DATA we define our own type for WM_COPYDATA message
+const WMCOPYDATA_SINGLE_INSTANCE_DATA = 1542
+
+func SendMessage(hwnd w32.HWND, data string) {
+	arrUtf16, _ := syscall.UTF16FromString(data)
+
+	pCopyData := new(COPYDATASTRUCT)
+	pCopyData.dwData = WMCOPYDATA_SINGLE_INSTANCE_DATA
+	pCopyData.cbData = uint32(len(arrUtf16)*2 + 1)
+	pCopyData.lpData = uintptr(unsafe.Pointer(windows.StringToUTF16Ptr(data)))
+
+	w32.SendMessage(hwnd, w32.WM_COPYDATA, 0, uintptr(unsafe.Pointer(pCopyData)))
+}
+
+// SetupSingleInstance single instance Windows app
+func SetupSingleInstance(uniqueId string) {
+	id := "wails-app-" + uniqueId
+
+	className := id + "-sic"
+	windowName := id + "-siw"
+	mutexName := id + "sim"
+
+	_, err := windows.CreateMutex(nil, false, windows.StringToUTF16Ptr(mutexName))
```

Comment:
I believe we need to also call `CloseHandle` for the mutex on shutdown

Reply:
If I understood this docs correctly https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-createmutexw

> Use the [CloseHandle](https://learn.microsoft.com/en-us/windows/desktop/api/handleapi/nf-handleapi-closehandle) function to close the handle. The system closes the handle automatically when the process terminates. The mutex object is destroyed when its last handle has been closed.

Windows shutdown mutexes if process terminates. So it doesn't require explicitly closing

Reply:
I guess I misread something, somewhere 😬